easybuild/toolchains/compiler/craype.py

```diff
@@ -103,6 +103,31 @@ def __init__(self, *args, **kwargs):
         # copy unique option map, since we fiddle with it later
         self.COMPILER_UNIQUE_OPTION_MAP = copy.deepcopy(self.COMPILER_UNIQUE_OPTION_MAP)
 
+    # this is specific to the semantics of craype-modules. There are craype modules for steering the compiler's backend code gen,
+    # while others define run time behaviour like the use of hugepages.    
+    def load_or_swap_module(self, mod_name_prefix, mod_name):
+	self.log.debug("Using load or swap")
+        if self.modules_tool.exist([mod_name], skip_avail=True)[0]:
+            if any(mod.startswith(mod_name_prefix) for mod in self.modules_tool.loaded_modules()):
+                loaded_mods_same_prefix = [mod for mod in self.modules_tool.loaded_modules() if
+                                      mod.startswith(mod_name_prefix)]
+                self.log.debug("Loaded modules of same prefix %s " % loaded_mods_same_prefix)
+                for loaded_mod in loaded_mods_same_prefix:
```

Comment:
the `for` loop doesn't make sense, you can only `swap` once...

so, I think you need to check how many loaded modules with the same prefix there are; if there's more than one, something is very wrong...